vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -316,29 +328,50 @@ and
                 projectContext.RemoveSourceFile(file)
                 updated <- true
         
-        let updatedRefs = site.AssemblyReferences() |> hashSetIgnoreCase
+        (*
```

Comment:
I may be wrong but IIRC project updates must run on UI thread, contrary to what some comments in Roslyn say.